test/resources/controls/aws_parallelcluster_install/mysql_client_spec.rb

```diff
@@ -16,20 +16,22 @@
   if os.redhat?
     mysql_packages.concat %w(mysql-community-client-plugins mysql-community-common
        mysql-community-devel mysql-community-libs mysql-community-libs-compat)
-  elsif os.debian?
-    if os.release == '18.04'
-      mysql_packages.concat %w(libmysqlclient-dev libmysqlclient20)
-    else
-      mysql_packages.concat %w(libmysqlclient-dev libmysqlclient21)
-    end
+  elsif os_properties.ubuntu1804?
+    mysql_packages.concat %w(libmysqlclient-dev libmysqlclient20)
+  elsif os_properties.ubuntu2004?
+    mysql_packages.concat %w(libmysqlclient-dev libmysqlclient21)
   else
     describe "unsupported OS" do
       pending "support for #{os.name}-#{os.release} needs to be implemented"
     end
   end
+
+  ubuntu = os_properties.ubuntu?
+
   mysql_packages.each do |pkg|
     describe package(pkg) do
       it { should be_installed }
+      its('version') { should match /^8.0.31-/ } unless ubuntu
```

Comment:
I think that this could just be inlined:

```suggestion
      its('version') { should match /^8.0.31-/ } unless os_properties.ubuntu?
```

or we could even use the existing check on Debian family which we typically use to discriminate changes that apply to the ubuntu OSs:

```suggestion
      its('version') { should match /^8.0.31-/ } unless os_properties.debian_family?
```

Reply:
Unfortunately, it can't: `os_properties` is not available in that context. This is why I used a variable.

I could use `os.debian?` instead of `os_properties.ubuntu?` since right now they are equivalent, however I preferred a narrower condition, which we can broaden in the future.